src/main/java/elements/Teleporter.java

```diff
@@ -1,18 +1,18 @@
 package elements;
-import game.Player;
 import game.PlayerState;
 import utils.Point2D;
 
-public class Teleporter extends Element implements Interactable, Teleportable{
+public class Teleporter extends Element implements Interactable{
 
     public Teleporter (String sprite, Point2D pos) {
         super(sprite, pos);
     }
 
     @Override
     public PlayerState changePlayerState(PlayerState playerState) {
+        playerState.setPos(this.getPos());
```

Comment:
You are setting the player to teleport to the position of the teleporter. But the player is already standing on the teleporter! Add an instance variable called "teleport location or something"

Reply:
Ok, I'll try that, I thought I had specified where the player will go in the ObjectManager, the modifyPlayerState method.

Reply:
This is where the player state is being modified, so it is impossible/incorrect for the playerstate to change elsewhere.

Reply:
resolved in commit 41df858.